### PR TITLE
samDeploy - ability to override cloudformation capabilities

### PR DESCRIPTION
**Pull request by Guslington**

### CODE REVIEW

File: vars/samDeploy.groovy [hunk 1]
```diff
@@ -44,6 +49,11 @@ def call(body) {
   if (config.noFailOnEmptyChangeset != null && config.noFailOnEmptyChangeset == "true") {
     options = options.concat(" --no-fail-on-empty-changeset")
   }
+ 
+  def capabilities = config.get('capabilities', ['CAPABILITY_IAM'])
+  if (capabilities) {
```
Comment (Samseppiol) [inline vars/samDeploy.groovy:54]: Is the if condition necessary when there is a default, therefore always defined? What would be the use case for not specifying capabilities?

File: vars/samDeploy.groovy [hunk 2]
```diff
@@ -15,6 +15,11 @@
    parameters: [
      'ENVIRONMENT_NAME' : 'dev',
    ],
+   capabilities: [ # define cloudfromation capabilities required by the stack or set value to false to disable the default capabilities
```
Comment (Samseppiol) [inline vars/samDeploy.groovy:18]: cloudfromation > cloudformation

File: vars/samDeploy.groovy (hunk 1 above)
Comment (Guslington) [reply vars/samDeploy.groovy:54]: the use case here is to set the value of capabilities to false to disable the default capabilities

**Review by Samseppiol — APPROVED** [review]
lgtm

**Pull request merged**